src/branch_and_bound.rs

```diff
@@ -7,14 +7,28 @@ use std::convert::TryInto;
 pub struct OutPointValue(pub OutPoint, pub u64);
 
 pub struct BranchAndBound {
+    /// The spending target to be reached. Note that this variable should hold only the sum of the
```

Comment:
Nit: This PR would be easier to review if the commit adding these comments were moved right after the commit that moved the previous code here. Then a reviewer would get the helpful explanations early in the review rather than in the last commit. :)